main/core/Converter/CurrentUserConverter.php

```
@@ -65,6 +83,15 @@ public function apply(Request $request, ParamConverter $configuration)
      */
     public function supports(ParamConverter $configuration)
     {
-        return $configuration->getConverter() === 'current_user';
+        if (!$configuration instanceof ParamConverter) {
+            return false;
+        }
+
+        $options = $configuration->getOptions();
+        if (isset($options['allowAnonymous']) && !is_bool($options['allowAnonymous'])) {
+            return false;
```

Having a non-boolean value for `allowAnonymous` is a configuration error IMO. Returning false means the error is silenced and the converter completely bypassed. An exception (here or in `apply`) would make more sense.